### PR TITLE
validation pipeline for the new beta Noble stemcell

### PR DESCRIPTION
**Pull request by dimivel**
### WHAT is this change about?
This is a PR for "Noble" stemcell validation. Adds a pipeline to deploy and run tests against the "Noble" stemcell.

- Requires a noble stemcell to have been uploaded to any given CF-D
environment.

> _Please describe the change._

### What customer problem is being addressed? Use customer persona to define the problem e.g. Alana is unable to...

Operators and maintainers want to start testing CF-Deployment with the NOble stemcell.

### Has a cf-deployment including this change passed [cf-acceptance-tests](https://github.com/cloudfoundry/cf-acceptance-tests)?

- [ ] YES
- [ ] NO
- [x] N/A

### Does this PR introduce a breaking change? Please take a moment to read through the examples before answering the question.

- [ ] YES - please choose the category from below. Feel free to provide additional details.
- [x] NO

### How should this change be described in cf-deployment release notes?

N/A

> _Something brief that conveys the change and is written with the **persona (Alana, Cody...)** in mind. See [previous release notes](https://github.com/cloudfoundry/cf-deployment/releases) for examples._

### Does this PR introduce a new BOSH release into the base cf-deployment.yml manifest or any ops-files?

- [ ] YES - please specify
- [x] NO

### Does this PR make a change to an experimental or GA'd feature/component?

- [ ] experimental feature/component
- [ ] GA'd feature/component
- [x] N/A

### Please provide Acceptance Criteria for this change?

> _Please specify either bosh cli or cf cli commands for our team (and cf operators) to verify the changes._

Validate that the noble-stemcell pipeline remains green (it uses the pipeline yml included in this PR) and is doing the intended work to test the noble stemcell.

### What is the level of urgency for publishing this change?

- [ ] **Urgent** - unblocks current or future work
- [x] **Slightly Less than Urgent**

### Tag your pair, your PM, and/or team!

> _It's helpful to tag a few other folks on your team or your team alias in case we need to follow up later._

### CODE REVIEW

File: ci/pipelines/noble-stemcell.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+## Triggers
+
+This pipeline is automatically triggered when new Noble stemcells are published, or when a cf-d commit passes through the cf-deployment CI to be promoted to the `release-candidate` branch.
+
+## Cleanup
+
+If the pipeline succeeds, then it will cleanup after itself.
+
+If the pipeline fails then there will be an orphaned Toolsmiths environment. This is by design so that we can examine the pipeline to determine the causes of any failure. In this situation, an environment can be manually unclaimed with the `unclaim-env-manual` job.
```
Comment (jochenehret) [inline ci/pipelines/noble-stemcell.md:13]: We don't use Toolsmiths environments. That was copied from an older pipeline, I assume?

File: ci/pipelines/noble-stemcell.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# noble-stemcell
+
+Test cf-d on the Ubuntu Noble stemcell.
+
+## Triggers
+
+This pipeline is automatically triggered when new Noble stemcells are published, or when a cf-d commit passes through the cf-deployment CI to be promoted to the `release-candidate` branch.
```
Comment (jochenehret) [inline ci/pipelines/noble-stemcell.md:7]: Maybe mention the stemcell repository that we use as trigger: https://bosh.io/stemcells/#ubuntu-noble

**Review by jochenehret — CHANGES_REQUESTED** [review]
Pipeline looks good, just update the documentation.

File: ci/pipelines/noble-stemcell.md (hunk 2 above)
Comment (dimivel) [reply ci/pipelines/noble-stemcell.md]: Done.

File: ci/pipelines/noble-stemcell.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+This pipeline is automatically triggered when new Noble stemcells are published to https://bosh.io/stemcells/#ubuntu-noble repository, or when a cf-d commit passes through the cf-deployment CI to be promoted to the `release-candidate` branch.
+
+## Cleanup
+
+If the pipeline succeeds, then it will clean up the CF BOSH deployment after itself.
+
+## Pipeline Management
+
+This pipeline is managed directly by the `ci/pipelines/noble-stemcell.yml` file and the `ci/configure` script. To update the pipeline, run `ci/configure nobleF-stemcell`.
```
Comment (jochenehret) [inline ci/pipelines/noble-stemcell.md:15]: Small typo: "nobleF"

**Review by jochenehret — APPROVED** [review]

File: units/tests/experimental_test/operations.yml [hunk 4]
```diff
@@ -85,3 +85,4 @@ disable-tls-tcp-routing-windows-stage-2-route-emitter.yml:
   - ../windows2019-cell.yml
   - enable-nginx-routing-integrity-windows2019.yml
   - disable-tls-tcp-routing-windows-stage-1-unproxied-ports.yml
+use-noble-stemcell: {}
```
Comment (ctlong) [inline units/tests/experimental_test/operations.yml:88]: I think this should actually have the `.yml` ending, e.g. `use-noble-stemcell.yml`

**Review by ctlong — CHANGES_REQUESTED** [review]

File: units/tests/experimental_test/operations.yml (hunk 4 above)
Comment (dimivel) [reply units/tests/experimental_test/operations.yml]: Done.

File: ci/pipelines/noble-stemcell.md (hunk 3 above)
Comment (dimivel) [reply ci/pipelines/noble-stemcell.md]: Done.

**Review by jochenehret — APPROVED** [review]

**Comment by jochenehret** [comment]
The filename in the README is not correct:
```
    --- FAIL: TestExperimental/ensure_use-noble-stemcell.yml_is_in_README (0.00s)
        operations_test.go:20: use-noble-stemcell.yml file was not found in README
    --- FAIL: TestExperimental/ensure_use-noble.yml_from_README_exists (0.00s)
        operations_test.go:20: use-noble.yml file from README doesn't exist
```
I noticed, but I wasn't aware that this fails the unit tests...

**Review by jochenehret — APPROVED** [review]

**Review by ctlong — DISMISSED** [review]

**Review by jochenehret — APPROVED** [review]
Pipeline has been uploaded to Concourse with the last changes.

**Pull request merged**